src/Core/Hook/Provider/GridDefinitionHookByServiceIdsProvider.php

```diff
@@ -61,10 +61,10 @@ public function __construct(ContainerInterface $container)
     /**
      * {@inheritdoc}
      */
-    public function getHookNames(array $gridDefinitionServiceIds)
+    public function getHookNames(array $gridDefinitionServiceIds, SymfonyStyle $io)
```

Comment:
I'm not sure it's the right approach to couple this service to `SymfonyStyle`
Especially through an argument, at  worst it should be injected in the constructor rather than provided via a parameter
This would avoid breaking/twisting the interface as well (see other comment)

And ideally it would be better to inject a `LoggerInterface` which is more generic, you even use a `NullLogger` and keep this class a bit more independent

Reply:
Agree, don't use SymfonyStyle in services :+1: